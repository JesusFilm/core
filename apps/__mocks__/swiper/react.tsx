import React from 'react'

export const Swiper = ({ children, 'data-testid': dataTestId }) => (
  <div data-testid={dataTestId}>{children}</div>
)
export const SwiperSlide = ({ children, 'data-testid': dataTestId }) => (
  <div data-testid={dataTestId}>{children}</div>
)
export const SwiperClass = {}
