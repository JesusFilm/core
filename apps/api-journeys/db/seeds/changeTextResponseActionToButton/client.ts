import { PrismaClient } from '.prisma/api-journeys-client'

const prisma = new PrismaClient()
export default prisma
