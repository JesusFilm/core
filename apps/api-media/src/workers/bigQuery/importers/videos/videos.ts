import { Logger } from 'pino'
import { z } from 'zod'

import { Prisma, VideoLabel } from '.prisma/api-media-client'

import { prisma } from '../../../../lib/prisma'
import { slugify } from '../../../../lib/slugify'
import { parse, parseMany, processTable } from '../../importer'

const videoSchema = z.object({
  id: z.string(),
  label: z.string().transform<VideoLabel>((value) => {
    switch (value) {
      case 'short':
        return VideoLabel.shortFilm
      case 'feature':
        return VideoLabel.featureFilm
      case 'behind_the_scenes':
        return VideoLabel.behindTheScenes
      case 'segments':
        return VideoLabel.segment
      default:
        return value as VideoLabel
    }
  }),
  primaryLanguageId: z.number().transform(String),
  slug: z.string(),
  childIds: z
    .string()
    .nullable()
    .transform((value) =>
      value == null ? [] : value.substring(1, value.length - 1).split(',')
    )
    .transform((value) => value.filter((id) => id.length > 0))
})

type Video = z.infer<typeof videoSchema>

let videoIds: string[] = []

export function setVideoIds(videos: Array<{ id: string }>): void {
  videoIds = videos.map(({ id }) => id)
}

export function getVideoIds(): string[] {
  return videoIds
}

let videoSlugs: Record<string, string> = {}

export function setVideoSlugs(
  videos: Array<{ id: string; slug: string | null }>
): void {
  videoSlugs = {}
  videos.forEach(({ id, slug }) => {
    if (slug != null) videoSlugs[slug] = id
  })
}

export function getVideoSlugs(): Record<string, string> {
  return videoSlugs
}

export async function importVideos(logger?: Logger): Promise<() => void> {
  await processTable(
    'jfp-data-warehouse.jfp_mmdb_prod.core_video_arclight_data',
    importOne,
    importMany,
    true,
    logger
  )

  const videos = await prisma.video.findMany({
    select: { id: true, slug: true }
  })
  setVideoIds(videos)
  setVideoSlugs(videos)

  return () => {
    setVideoIds([])
    setVideoSlugs([])
  }
}

function transform(
  video: Video
): Prisma.VideoCreateInput & { id: string; slug: string } {
  return {
    ...video,
    slug: slugify(video.id, video.slug, videoSlugs),
    noIndex: false
  }
}

export async function importOne(row: unknown): Promise<void> {
  const video = parse(videoSchema, row)
  const input = transform(video)
  await prisma.video.upsert({
    where: { id: video.id },
    update: input,
    create: input
  })
}

export async function importMany(rows: unknown[]): Promise<void> {
  const { data: videos, inValidRowIds } = parseMany(videoSchema, rows)

  if (videos.length !== rows.length)
    throw new Error(`some rows do not match schema: ${inValidRowIds.join(',')}`)

  const transformedVideos = videos
    .map((input) => transform(input))
    .filter(({ id }) => !getVideoIds().includes(id))

  await prisma.video.createMany({
    data: transformedVideos,
    skipDuplicates: true
  })
}
