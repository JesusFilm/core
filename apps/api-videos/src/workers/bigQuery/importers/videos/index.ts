export { importVideos, getVideoIds, getVideoSlugs } from './videos'
