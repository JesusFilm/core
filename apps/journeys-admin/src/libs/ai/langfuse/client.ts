import { LangfuseWeb } from 'langfuse'

export const langfuseWeb = new LangfuseWeb({
  publicKey: process.env.NEXT_PUBLIC_LANGFUSE_PUBLIC_KEY ?? '',
  baseUrl: process.env.NEXT_PUBLIC_LANGFUSE_BASE_URL ?? ''
})
