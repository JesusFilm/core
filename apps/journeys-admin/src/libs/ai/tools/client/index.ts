// import { generateImage } from './generateImage'
import { clientRedirectUserToEditor } from './redirectUserToEditor'
import { clientRequestForm } from './requestForm'
import { clientSelectImage } from './selectImage'
import { clientSelectVideo } from './selectVideo'

export const tools = {
  clientSelectImage,
  clientSelectVideo,
  // TODO: Uncomment this when we have solved image upload issues
  // generateImage,
  clientRedirectUserToEditor,
  clientRequestForm
}
