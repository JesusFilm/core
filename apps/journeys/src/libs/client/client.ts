import { ApolloClient, InMemoryCache } from '@apollo/client'

const client = new ApolloClient({
  uri: 'http://localhost:4000',
  cache: new InMemoryCache()
})

export default client
