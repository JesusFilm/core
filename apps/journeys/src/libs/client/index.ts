import client from './client'

export default client
