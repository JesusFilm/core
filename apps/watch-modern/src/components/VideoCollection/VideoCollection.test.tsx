import { MockedProvider } from '@apollo/client/testing'
import { fireEvent, render, screen, waitFor } from '@testing-library/react'

import { VideoCollection } from './index'

import { defaultGetVideosMocks, errorGetVideosMocks } from '@/test/apolloMocks'
import { mockVideos } from '@/test/fixtures/videos'

// Mock window.location.href
const mockLocation = {
  href: ''
}
Object.defineProperty(window, 'location', {
  value: mockLocation,
  writable: true
})

describe('VideoCollection', () => {
  beforeEach(() => {
    mockLocation.href = ''
  })

  it('renders correctly', () => {
    const { container } = render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('displays the section title', () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    const title = screen.getByText('VIDEO BIBLE COLLECTION')
    expect(title).toBeInTheDocument()
  })

  it('displays the main heading', () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    const heading = screen.getByText('Video Gospel in every style and language')
    expect(heading).toBeInTheDocument()
  })

  it('displays the description', () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    const description = screen.getByText(/Experience the life of Jesus through authentic, faithful films/)
    expect(description).toBeInTheDocument()
  })

  it('displays video cards when data is loaded', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={true}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // First check that loading state appears
    const loadingElements = screen.getAllByTestId('loading-skeleton')
    expect(loadingElements.length).toBeGreaterThan(0)
    
    // Wait for video cards to load (either from mocks or real API)
    const videoCards = await screen.findAllByRole('button')
    expect(videoCards.length).toBeGreaterThan(0)
    
    // Check that at least one video card has proper structure
    const firstCard = videoCards[0]
    expect(firstCard).toHaveAttribute('tabIndex', '0')
    expect(firstCard).toHaveAttribute('aria-label')
  })

  it('navigates to video detail page when clicked', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video cards to load
    const videoCards = await screen.findAllByRole('button')
    expect(videoCards.length).toBeGreaterThan(0)
    
    fireEvent.click(videoCards[0])
    
    // Check that navigation was attempted (slug may vary)
    expect(mockLocation.href).toMatch(/\/watch\//)
  })

  it('navigates to video detail page when Enter key is pressed', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video cards to load
    const videoCards = await screen.findAllByRole('button')
    expect(videoCards.length).toBeGreaterThan(0)
    
    fireEvent.keyDown(videoCards[0], { key: 'Enter' })
    
    // Check that navigation was attempted
    expect(mockLocation.href).toMatch(/\/watch\//)
  })

  it('navigates to video detail page when Space key is pressed', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video cards to load
    const videoCards = await screen.findAllByRole('button')
    expect(videoCards.length).toBeGreaterThan(0)
    
    fireEvent.keyDown(videoCards[0], { key: ' ' })
    
    // Check that navigation was attempted
    expect(mockLocation.href).toMatch(/\/watch\//)
  })

  it('has proper accessibility attributes', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video cards to load
    const videoCards = await screen.findAllByRole('button')
    expect(videoCards.length).toBeGreaterThan(0)
    
    videoCards.forEach((card) => {
      expect(card).toHaveAttribute('tabIndex', '0')
      expect(card).toHaveAttribute('aria-label')
    })
  })

  it('displays the mission text', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video cards to load
    await screen.findAllByRole('button')
    const missionText = screen.getByText(/Our mission/)
    expect(missionText).toBeInTheDocument()
  })

  it('has responsive grid layout', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video cards to load
    const videoCards = await screen.findAllByRole('button')
    expect(videoCards.length).toBeGreaterThan(0)
    
    const grid = videoCards[0]?.closest('.grid')
    expect(grid).toHaveClass('grid-cols-1', 'sm:grid-cols-2', 'lg:grid-cols-3', 'xl:grid-cols-4')
  })

  it('shows loading state initially', () => {
    render(
      <MockedProvider mocks={[]} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Should show loading skeleton
    const loadingElements = screen.getAllByTestId('loading-skeleton')
    expect(loadingElements.length).toBeGreaterThan(0)
  })

  it('shows error state when query fails', async () => {
    render(
      <MockedProvider mocks={errorGetVideosMocks} addTypename={false}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for error state
    const errorTitle = await screen.findByText('Error Loading Films')
    expect(errorTitle).toBeInTheDocument()
    
    const errorMessage = screen.getByText('Unable to load video content. Please try again later.')
    expect(errorMessage).toBeInTheDocument()
  })

  it('displays video titles correctly', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={true}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video cards to load
    await waitFor(() => {
      const videoCards = screen.getAllByRole('button')
      expect(videoCards.length).toBeGreaterThan(0)
    })
    
    // Check that video titles are displayed as headings
    const videoTitles = screen.getAllByRole('heading', { level: 3 })
    expect(videoTitles.length).toBeGreaterThan(0)
    
    // Check that at least one title is not empty
    const hasNonEmptyTitle = videoTitles.some(title => title.textContent && title.textContent.trim() !== '')
    expect(hasNonEmptyTitle).toBe(true)
  })

  it('displays video snippets correctly', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={true}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video content to load
    await waitFor(() => {
      const videoCards = screen.getAllByRole('button')
      expect(videoCards.length).toBeGreaterThan(0)
    })
    
    // Check that video cards have content
    const videoCards = screen.getAllByRole('button')
    expect(videoCards.length).toBeGreaterThan(0)
  })

  it('displays video durations correctly', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={true}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video content to load
    await waitFor(() => {
      const videoCards = screen.getAllByRole('button')
      expect(videoCards.length).toBeGreaterThan(0)
    })
    
    // Check that duration information is displayed
    const durationElements = screen.getAllByText(/MIN|HR/)
    expect(durationElements.length).toBeGreaterThan(0)
  })

  it('displays language counts correctly', async () => {
    render(
      <MockedProvider mocks={defaultGetVideosMocks} addTypename={true}>
        <VideoCollection />
      </MockedProvider>
    )
    
    // Wait for video content to load
    await waitFor(() => {
      const videoCards = screen.getAllByRole('button')
      expect(videoCards.length).toBeGreaterThan(0)
    })
    
    // Check that language count information is displayed
    const languageElements = screen.getAllByText(/LANGUAGES/)
    expect(languageElements.length).toBeGreaterThan(0)
  })
}) 