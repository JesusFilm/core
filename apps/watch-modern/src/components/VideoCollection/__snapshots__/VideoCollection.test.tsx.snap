// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCollection renders correctly 1`] = `
<div>
  <section
    class="min-h-screen bg-slate-950 bg-gradient-to-tr from-blue-950/10 via-purple-950/10 to-[#91214A]/90 py-16 scroll-snap-start-always text-white relative"
  >
    <div
      class="max-w-[1920px] mx-auto px-4 sm:px-8 lg:px-20 py-12 relative z-10"
    >
      <div
        class="flex justify-between items-start mb-12"
      >
        <div
          class="max-w-5xl"
        >
          <p
            class="text-stone-200/80 text-sm tracking-[0.3em] uppercase mb-8 font-medium text-[rgba(255,255,255,0.6)]"
          >
            VIDEO BIBLE COLLECTION
          </p>
          <h2
            class="text-3xl xl:text-4xl 2xl:text-5xl font-bold leading-[0.95] tracking-tight text-white mb-6"
          >
            Video Gospel in every style and language
          </h2>
          <p
            class="text-stone-100/90 text-xl leading-relaxed max-w-3xl text-[rgba(255,255,255,0.9)]"
          >
            Experience the life of Jesus through authentic, faithful films translated into thousands of languages worldwide.
          </p>
        </div>
      </div>
    </div>
    <div
      class="max-w-[1920px] mx-auto px-4 sm:px-8 lg:px-20 relative z-10 mb-12"
    >
      <div
        class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-16 sm:gap-6"
      >
        <div
          class="w-[75%] sm:w-auto mx-auto sm:mx-0"
        >
          <div
            class="relative aspect-[2/3] rounded-xl overflow-hidden mb-3 bg-slate-800 animate-pulse"
          >
            <div
              class="absolute inset-0 bg-slate-700"
            />
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="flex items-center gap-2 text-stone-200/80 text-xs uppercase tracking-wider mb-2"
            >
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-16"
                data-testid="loading-skeleton"
              />
              <span
                class="w-1 h-1 bg-stone-200/60 rounded-full"
              />
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-20"
                data-testid="loading-skeleton"
              />
            </div>
            <div
              class="h-6 bg-slate-700 rounded animate-pulse mb-2"
              data-testid="loading-skeleton"
            />
            <div
              class="h-4 bg-slate-700 rounded animate-pulse"
              data-testid="loading-skeleton"
            />
          </div>
        </div>
        <div
          class="w-[75%] sm:w-auto mx-auto sm:mx-0"
        >
          <div
            class="relative aspect-[2/3] rounded-xl overflow-hidden mb-3 bg-slate-800 animate-pulse"
          >
            <div
              class="absolute inset-0 bg-slate-700"
            />
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="flex items-center gap-2 text-stone-200/80 text-xs uppercase tracking-wider mb-2"
            >
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-16"
                data-testid="loading-skeleton"
              />
              <span
                class="w-1 h-1 bg-stone-200/60 rounded-full"
              />
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-20"
                data-testid="loading-skeleton"
              />
            </div>
            <div
              class="h-6 bg-slate-700 rounded animate-pulse mb-2"
              data-testid="loading-skeleton"
            />
            <div
              class="h-4 bg-slate-700 rounded animate-pulse"
              data-testid="loading-skeleton"
            />
          </div>
        </div>
        <div
          class="w-[75%] sm:w-auto mx-auto sm:mx-0"
        >
          <div
            class="relative aspect-[2/3] rounded-xl overflow-hidden mb-3 bg-slate-800 animate-pulse"
          >
            <div
              class="absolute inset-0 bg-slate-700"
            />
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="flex items-center gap-2 text-stone-200/80 text-xs uppercase tracking-wider mb-2"
            >
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-16"
                data-testid="loading-skeleton"
              />
              <span
                class="w-1 h-1 bg-stone-200/60 rounded-full"
              />
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-20"
                data-testid="loading-skeleton"
              />
            </div>
            <div
              class="h-6 bg-slate-700 rounded animate-pulse mb-2"
              data-testid="loading-skeleton"
            />
            <div
              class="h-4 bg-slate-700 rounded animate-pulse"
              data-testid="loading-skeleton"
            />
          </div>
        </div>
        <div
          class="w-[75%] sm:w-auto mx-auto sm:mx-0"
        >
          <div
            class="relative aspect-[2/3] rounded-xl overflow-hidden mb-3 bg-slate-800 animate-pulse"
          >
            <div
              class="absolute inset-0 bg-slate-700"
            />
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="flex items-center gap-2 text-stone-200/80 text-xs uppercase tracking-wider mb-2"
            >
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-16"
                data-testid="loading-skeleton"
              />
              <span
                class="w-1 h-1 bg-stone-200/60 rounded-full"
              />
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-20"
                data-testid="loading-skeleton"
              />
            </div>
            <div
              class="h-6 bg-slate-700 rounded animate-pulse mb-2"
              data-testid="loading-skeleton"
            />
            <div
              class="h-4 bg-slate-700 rounded animate-pulse"
              data-testid="loading-skeleton"
            />
          </div>
        </div>
        <div
          class="w-[75%] sm:w-auto mx-auto sm:mx-0"
        >
          <div
            class="relative aspect-[2/3] rounded-xl overflow-hidden mb-3 bg-slate-800 animate-pulse"
          >
            <div
              class="absolute inset-0 bg-slate-700"
            />
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="flex items-center gap-2 text-stone-200/80 text-xs uppercase tracking-wider mb-2"
            >
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-16"
                data-testid="loading-skeleton"
              />
              <span
                class="w-1 h-1 bg-stone-200/60 rounded-full"
              />
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-20"
                data-testid="loading-skeleton"
              />
            </div>
            <div
              class="h-6 bg-slate-700 rounded animate-pulse mb-2"
              data-testid="loading-skeleton"
            />
            <div
              class="h-4 bg-slate-700 rounded animate-pulse"
              data-testid="loading-skeleton"
            />
          </div>
        </div>
        <div
          class="w-[75%] sm:w-auto mx-auto sm:mx-0"
        >
          <div
            class="relative aspect-[2/3] rounded-xl overflow-hidden mb-3 bg-slate-800 animate-pulse"
          >
            <div
              class="absolute inset-0 bg-slate-700"
            />
          </div>
          <div
            class="space-y-1"
          >
            <div
              class="flex items-center gap-2 text-stone-200/80 text-xs uppercase tracking-wider mb-2"
            >
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-16"
                data-testid="loading-skeleton"
              />
              <span
                class="w-1 h-1 bg-stone-200/60 rounded-full"
              />
              <div
                class="h-4 bg-slate-700 rounded animate-pulse w-20"
                data-testid="loading-skeleton"
              />
            </div>
            <div
              class="h-6 bg-slate-700 rounded animate-pulse mb-2"
              data-testid="loading-skeleton"
            />
            <div
              class="h-4 bg-slate-700 rounded animate-pulse"
              data-testid="loading-skeleton"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
